### DISCUSSION TITLE
Developer Experience

### DISCUSSION THREAD

**Discussion by TheSavageDev** [DevVet/codeChallengeAPI#14] [Ideas]
We should decide on coding standards before we get too far.

What are we going to use for linting, formatting, testing framework, etc.

I would suggest

- SwaggerUI
- eslint
- Prettier if you want, but would make it a better DX for others because they can code like they code and prettier will take over the formatting.
- TravisCI for CI...
- Jest and Cypress for unit and E2E testing
- .nvmrc for node versioning

**Comment by TheSavageDev** [comment]
.nvmrc added.

**Comment by bmw2621** [comment]
Default indentation for prettier is 2 spaces as you mentioned in your review.  We can change it back.  Do we need a `.prettierrc`
